Resource scaffolding script v2 (#530)
hashicorp/go-tfe: docs/CONTRIBUTING.md

## Running the Linters Locally

1. Ensure you have [installed golangci-lint](https://golangci-lint.run/usage/install/#local-installation)

> 2. From the CLI, run `make lint`

Reviewer: 💯 This is nice!

**Note**: After your PR has been merged, and the feature either reaches general availability, you should remove the `skipIfBeta()` flag.

## Adding New Endpoints

### Scaffolding a Resource

When creating a new resource you can use the helper script `generate_resource` to quickly setup boilerplate code for adding a new set of endpoints related to that resource:

#### Running the script directly
```sh
cd ./scripts/generate_resource
go run . example_resource
```

#### Running the Makefile target `generate`
```sh

> RESOURCE=example_resource make generate

Reviewer: This command is different from what is in the PR description. Do both options work?

Author: Just updated!

```

### Guidelines for Adding New Endpoints

* An interface should cover one RESTful resource, which sometimes involves two or more endpoints.
* We require that each resource interface provides compile-time proof that it has been implemented.
* You'll need to add an integration test that covers each method of the resource's interface.
* Option structs serve as a proxy for either passing query params or request bodies:
    - `ListOptions` and `ReadOptions` are values passed as query parameters.
    - `CreateOptions` and `UpdateOptions` represent the request body.
* URL parameters should be defined as method parameters.
* Any resource specific errors must be defined in `errors.go`

> Here is a more comprehensive example of how a resource will look like when implemented. The helper script `generate_resource` will generate a subset of this example, focusing only on the core details that are required across all resources in go-tfe.

Reviewer: ```suggestion
Here is a more comprehensive example of what a resource looks like when implemented. The helper script `generate_resource` generates a subset of this example, focusing only on the core details that are required across all resources in go-tfe.
```
